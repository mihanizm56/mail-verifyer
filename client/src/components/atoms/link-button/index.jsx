import React from "react";
import { NavLink } from "react-router-dom";
import { OpenSansText } from "../../../components";
import classNames from "classnames";
import "./link-button.css";

export const LinkButton = ({ to, text, handleButtonClick }) => (
	<NavLink exact to={to} className={classNames("link-button")} onClick={handleButtonClick}>
		<OpenSansText text={text} isButton />
	</NavLink>
);

LinkButton.defaultProps = {
	handleButtonClick: () => console.log("default handleButtonClick"),
};
