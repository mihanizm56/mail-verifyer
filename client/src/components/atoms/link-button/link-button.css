.link-button {
    border: 1px solid black;
    border-radius: 15px;
    transition: .5s;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 300px;
    height: 70px;
}

.link-button__button {
    text-decoration: none;
}

.link-button:hover {
    background-color: var(--black);
    --color-button-text: #ffffff;
}

@media screen and (max-width: 767px) {}