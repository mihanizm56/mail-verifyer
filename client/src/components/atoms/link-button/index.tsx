import React, { memo } from "react";
import { NavLink } from "react-router-dom";
import classNames from "classnames";
import { OpenSansText } from "../../../components";
import "./link-button.css";

type LinkButtonProps = {
	to: string,
	text: string,
	handleButtonClick?: () => void,
};

export const LinkButton = memo(({ to, text, handleButtonClick }: LinkButtonProps) => (
	<NavLink exact to={to} className={classNames("link-button")} onClick={handleButtonClick}>
		<OpenSansText text={text} isButton />
	</NavLink>
));
