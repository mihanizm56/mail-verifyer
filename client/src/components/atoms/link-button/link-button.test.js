import React, { createElement } from "react";
import { shallow } from "enzyme";
import { LinkButton } from "../link-button";

describe("test LinkButton", () => {
	beforeEach(() => {
		jest.clearAllMocks();
	});

	describe("test render contract", () => {
		it("render snap with all props", () => {
			const props = {
				to: "/test",
				text: "test_lnk_button",
				handleButtonClick: jest.fn(),
			};
			const component = shallow(<LinkButton {...props} />);

			expect(component).toMatchSnapshot();
		});
		it("render NavLink component", () => {
			const props = {
				to: "/test",
				text: "test_lnk_button",
				handleButtonClick: jest.fn(),
			};
			const component = shallow(<LinkButton {...props} />);

			expect(component.find("NavLinkMock").length).toEqual(1);
		});
		it("render OpenSansText component", () => {
			const props = {
				to: "/test",
				text: "test_lnk_button",
				handleButtonClick: jest.fn(),
			};
			const component = shallow(<LinkButton {...props} />);

			expect(component.find("OpenSansTextMock").length).toEqual(1);
		});
	});
	describe("test props contract", () => {
		describe("set props to NavLink", () => {
			it("set exact prop", () => {
				const props = {
					to: "/test",
					text: "test_lnk_button",
					handleButtonClick: jest.fn(),
				};
				const component = shallow(<LinkButton {...props} />);
				const NavLink = component.find("NavLinkMock");
				const exactProp = NavLink.prop("exact");

				expect(exactProp).toBeTruthy();
			});
			it("set to (path) prop", () => {
				const props = {
					to: "/test",
					text: "test_lnk_button",
					handleButtonClick: jest.fn(),
				};
				const component = shallow(<LinkButton {...props} />);
				const NavLink = component.find("NavLinkMock");
				const toProp = NavLink.prop("to");

				expect(toProp).toEqual(props.to);
			});
			it("set to onClick prop", () => {
				const props = {
					to: "/test",
					text: "test_lnk_button",
					handleButtonClick: jest.fn(),
				};
				const component = shallow(<LinkButton {...props} />);
				const NavLink = component.find("NavLinkMock");
				const onClickProp = NavLink.prop("onClick");

				expect(onClickProp).toEqual(props.handleButtonClick);
			});
		});
		describe("set props to OpenSansText", () => {
			it("set text prop", () => {
				const props = {
					to: "/test",
					text: "test_lnk_button",
					handleButtonClick: jest.fn(),
				};
				const component = shallow(<LinkButton {...props} />);
				const OpenSansText = component.find("OpenSansTextMock");
				const textProp = OpenSansText.prop("text");

				expect(textProp).toEqual(props.text);
			});
			it("set isButton prop", () => {
				const props = {
					to: "/test",
					text: "test_lnk_button",
					handleButtonClick: jest.fn(),
				};
				const component = shallow(<LinkButton {...props} />);
				const OpenSansText = component.find("OpenSansTextMock");
				const isButtonProp = OpenSansText.prop("isButton");

				expect(isButtonProp).toBeTruthy();
			});
		});
	});
	describe("test callback contract", () => {
		it("test handleButtonClick goes when onClick on the button is done", () => {
			const props = {
				to: "/test",
				text: "test_lnk_button",
				handleButtonClick: jest.fn(),
			};
			const component = shallow(<LinkButton {...props} />);
			const NavLink = component.find("NavLinkMock");
			NavLink.simulate("click", {});

			expect(props.handleButtonClick).toHaveBeenCalled();
		});
	});
});
