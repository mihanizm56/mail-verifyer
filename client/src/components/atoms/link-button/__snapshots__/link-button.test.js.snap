// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LinkButton test render contract render snap with all props 1`] = `
<NavLinkMock
  className="link-button"
  exact={true}
  onClick={[MockFunction]}
  to="/test"
>
  <OpenSansTextMock
    isButton={true}
    text="test_lnk_button"
  />
</NavLinkMock>
`;
